core/quality/morqa_transport.go

```diff
@@ -50,14 +46,49 @@ func mapEventToMetric(event Event) *metrics.Event {
 	switch event.EventName {
 	case startupEventName:
 		return &metrics.Event{
-			IsProvider: false,
-			TargetId:   "0x1",
 			Metric: &metrics.Event_VersionPayload{
 				VersionPayload: &metrics.VersionPayload{
 					Version: event.Application.Version,
 				},
 			},
 		}
+	case sessionEventName:
+		context := event.Context.(sessionEventContext)
+		return &metrics.Event{
```

Comment:
add a method on sessionEventContext toEvent() and call it here to avoid this tedious block in the switch

Reply:
done